tests/unit/test_vault_tls.py

```diff
@@ -0,0 +1,414 @@
+        }
+        original_leader_state = self.harness.charm.unit.is_leader()
+        with self.harness.hooks_disabled():
+            self.harness.set_leader(is_leader=True)
+            secret_id = self.harness.add_model_secret(owner=self.app_name, content=content)
+            secret = self.harness.model.get_secret(id=secret_id)
+            secret.set_info(label=CA_CERTIFICATE_JUJU_SECRET_LABEL)
+            self.harness.set_leader(original_leader_state)
+        key_values = {"vault-ca-certificates-secret-id": secret_id}
+        self.harness.update_relation_data(
+            app_or_unit=self.app_name,
+            relation_id=relation_id,
+            key_values=key_values,
+        )
+
+    def _set_other_node_api_address_in_peer_relation(self, relation_id: int, unit_name: str):
+        """Set the other node api address in the peer relation."""
+        key_values = {"node_api_address": "http://5.2.1.9:8200"}
+        self.harness.update_relation_data(
+            app_or_unit=unit_name,
+            relation_id=relation_id,
+            key_values=key_values,
+        )
+
+    def _set_tls_access_certificate_relation(self):
+        """Set the peer relation and return the relation id."""
+        return self.harness.add_relation(
+            relation_name="tls-certificates-access", remote_app="some-tls-provider"
+        )
+
+    @patch("ops.model.Model.get_binding")
+    def test_given_not_leader_and_ca_not_set_when_configure_then_status_is_waiting(
+        self, patch_get_binding
+    ):
+        self.harness.add_storage(storage_name="certs", attach=True)
+        self.harness.set_can_connect(container=self.container_name, val=True)
+        self.harness.set_leader(is_leader=False)
+        peer_relation_id = self._set_peer_relation()
+        other_unit_name = f"{self.app_name}/1"
+        self.harness.add_relation_unit(
+            relation_id=peer_relation_id, remote_unit_name=other_unit_name
+        )
+        self._set_other_node_api_address_in_peer_relation(
+            relation_id=peer_relation_id, unit_name=other_unit_name
+        )
+        patch_get_binding.return_value = MockBinding(
+            bind_address="1.2.1.2", ingress_address="10.1.0.1"
+        )
+
+        self.harness.charm.on.config_changed.emit()
+
+        self.assertEqual(
+            self.harness.charm.unit.status,
+            WaitingStatus("Waiting for CA certificate to be set in peer relation"),
+        )
+
+    @patch("vault.Vault.enable_audit_device", new=Mock)
+    @patch("vault.Vault.is_active", new=Mock)
+    @patch("vault.Vault.audit_device_enabled", new=Mock)
+    @patch("vault.Vault.unseal", new=Mock)
+    @patch("vault.Vault.is_sealed", new=Mock)
+    @patch("vault.Vault.is_initialized", new=Mock)
+    @patch("vault.Vault.is_api_available", new=Mock)
+    @patch("ops.model.Container.push", new=Mock)
+    @patch("vault.Vault.is_raft_cluster_healthy", new=Mock)
+    @patch("ops.model.Model.get_binding")
```

Comment:
The `vault_tls` lib doesn't make any call to Vault, it doesn't make much sense to mock those. I understand that our tests don't really aim at the lib but at the charm implementing it but typically in those cases we write a dummy charm to test this. My concern here is that because of this we'll have to write the same tests for the machine charm.